.cursor/rules/golang.mdc

```diff
@@ -3,9 +3,110 @@ description: Golang
 globs: *.go
 alwaysApply: false
 ---
+---
+description: Golang
+globs: *.go
+alwaysApply: false
+---
 
 # Golang Rules
 
 - We use Ginkgo v2 & Gomega for testing
 - Do not remove focused tests
-- Do not remove existing documentation, unless they are being replaced with a better or more up to date one.
\ No newline at end of file
+- Do not remove existing documentation, unless they are being replaced with a better or more up to date one.
+
+## Efficient Go
```

Comment:
Can we add that to only "trigger" when performance is really needed? I want to prevent writing overly complicated code, that is unnecessarily performant. Something like "when the user explicitly requests this". Furthermore, I think applying this on our filesystem logic would probably give us a huge performance boost, so to fetch the entire S6 directory at the beginning of the loop, and then use it throughout the same tick.

Reply:
This already contains sections to advice the AI on when to use this